### ISSUE TITLE
Feature Request: Switch to allow mixins with content last in DeclarationOrder

### ISSUE THREAD

**Issue by Undistraction** [brigade/scss-lint#494]
Consider the following:

```
.Example {
  @include example(yes);
  width: 100%;
  
  .Icon {
    color: red;
  }

  @include bp(small-up) {
    @include example(no);
     width: 50%;

     .Icon {
       color: blue;
     }
  }
}
```

When using a mixin for rendering breakpoints, it makes more sense for it to come after any nested classes. Please supply a whitelist parameter for mixins that can come after everything else in the DeclarationOrder linter.

**Comment by lencioni** [comment]
> When using a mixin for rendering breakpoints, it makes more sense for it to come after any nested classes.

Why do you say that it makes more sense for it to come after nested classes? I believe that things should be ordered in how closely they relate to the root selector. The example you give would be better written as:

```scss
.Example {
  @include example(yes);
  width: 100%;
                           
  @include bp(small-up) {
    @include example(no);
    width: 50%;
  }
                           
  .Icon {
    color: red;
                           
    @include bp(small-up) {
      color: blue;
    }
  }
}
```

**Comment by Undistraction** [comment]
Well I suppose that's just a matter of personal preference. In your example you scatter state(breakpoint) down through children. In mine I prefer each state to exist in one place so I don't have to go looking for where a breakpoint occurs within a component. 

More importantly, without this feature, your example would be rendered:

```
.Example {
  @include example(yes);
  @include bp(small-up) {
    @include example(no);
    width: 50%;
  }


  .Icon {
    color: red;

    @include bp(small-up) {
      color: blue;
    }
  }
}
```

**Comment by lencioni** [comment]
Would you also put other state such as `:hover` after nested selectors?

**Comment by Undistraction** [comment]
Yes. Because otherwise you might be defining  the state of a nested Icon when a button is hovered before you define its default state. Obviously this isn't going to effect the outcome, but in terms of comprehension I think it's preferable.

**Comment by Anahkiasen** [comment]
I'd like that as well. Makes more sense to me to have special cases after default styles

**Comment by Undistraction** [comment]
To be honest, this isn't about whether my way is preferable to yours or vice-versa, just that people do things differently, often with equally valid, different reasons, so it would be great to support both approaches.

**Comment by lencioni** [comment]
Oh yeah, I totally agree. I just want to make sure that we thoroughly understand your problem before we solve it. :)

Right now [the declaration order is built into the linter](https://github.com/brigade/scss-lint/blob/6fb318effe4e10a6f67f8f7e3914b4519f414720/lib/scss_lint/linter/declaration_order.rb#L25) and is currently

```ruby
DECLARATION_ORDER = [
  Sass::Tree::ExtendNode,
  Sass::Tree::MixinNode,
  Sass::Tree::PropNode,
  MIXIN_WITH_CONTENT,
  Sass::Tree::RuleNode,
]
```

Would it satisfactorily solve your problem if that order was configurable?

**Comment by nenadjelovac** [comment]
> Would it satisfactorily solve your problem if that order was configurable?

Yes please! :)

**Comment by Undistraction** [comment]
That would be really great.

If we could treat mixins interchangeably with properties so that the name of a mixin could define its position, that would be even better ;)

**Comment by ads1018** [comment]
I prefer to declare parent selectors @includes with content. Currently, however, this throws a warning:

`.header__logo-container {
    @include transition(top $header-toggle-speed);
    background-color: $cobalt-blue;
    display: table-cell;

    &:hover {
        top: 35px;
    }

    @include media($large-screen-up) {
        height: 130px;
    }
}`

Is it possible to include the parent selector in declaration order? Also, is there a reason the linter prefers parent selectors to come after @includes with content? Perhaps I'm missing something.

**Comment by ste-m** [comment]
+1 for configurable order

**Comment by dbox** [comment]
+1 to configurable order.
My current ideal flow would be:
@extends, @includes without content, properties, nested rule sets, @includes wth content.

**Comment by jblandry** [comment]
+1 to configurable order.
My current ideal flow would be:
@extends, @includes without content, @includes with content, properties, nested rule sets.
